client/asset/dcr/rpcwallet.go

```diff
@@ -688,10 +688,19 @@ func (w *rpcWallet) LockAccount(ctx context.Context, acctName string) error {
 	return translateRPCCancelErr(err)
 }
 
-// UnlockAccount unlocks the specified account.
-// Part of the Wallet interface.
+// UnlockAccount unlocks the specified account or the wallet if account is not
+// encrypted. Part of the Wallet interface.
 func (w *rpcWallet) UnlockAccount(ctx context.Context, pw []byte, acctName string) error {
-	return translateRPCCancelErr(w.rpcClient.UnlockAccount(ctx, acctName, string(pw)))
+	var res *walletjson.AccountUnlockedResult
+	res, err := w.rpcClient.AccountUnlocked(ctx, acctName)
+	if err != nil {
+		return err
+	}
+	if res.Encrypted {
+		return translateRPCCancelErr(w.rpcClient.UnlockAccount(ctx, acctName, string(pw)))
```

Comment:
Thank you for restoring this logic.  I suppose it makes sense to restore it down here in the `rpcWallet` code instead of up at the `ExchangeWallet` level where it was.

Also, I can confirm that `(*rpcWallet).LockAccount` does have this `accountunlocked` check.